File: src/main/java/edu/marmara/model/Student.java
package edu.marmara.model;

public class Student extends Person {
    private Long studentId;
    private Integer yearEnrolled;
    private Schedule weeklySchedule;
    private Transcript transcript;
    private Advisor advisor;
    private Integer semester;


    public Student() {
    }

    public Student(Long studentId, Integer yearEnrolled, Schedule weeklySchedule, Transcript transcript, Advisor advisor, Integer semester) {
        this.studentId = studentId;
        this.yearEnrolled = yearEnrolled;
        this.weeklySchedule = weeklySchedule;
        this.transcript = transcript;
        this.advisor = advisor;
        this.semester = semester;
    }

    public void addCurrentlyTakenCourseToTranscript(Course course){
        this.transcript.addCurrentlyTakenCourse(course);

    }
    public void addNotTakenCourseToTranscript(Course course){
        this.transcript.addNotTakenCourse(course);
    }
    public void addFailedCoursesToTranscript(Course course){
        this.transcript.addFailedCourse(course);
    }

    public void removeFromTranscriptNotTakenCourse(Course course){
        this.transcript.removeNotTakenCourse(course);
    }
    public void removeFromTranscriptFailedCourse(Course course){
        this.transcript.removeFailedCourse(course);
    }

    public void addCourseToSchedule(Course course){
        this.weeklySchedule.addCourse(course);
    }
    public Double calculateGPA() {
        return this.transcript.calculateGPA();
    }

    @Override
    public String toString() {
        return "Student{" +
                "studentId=" + studentId +
                ", yearEnrolled=" + yearEnrolled +
                ", weeklySchedule=" + weeklySchedule +
                ", transcript=" + transcript +
                ", advisor=" + advisor +
                ", semester=" + semester +
                '}';
    }

    public Long getStudentId() {
        return studentId;
    }

    public void setStudentId(Long studentId) {
        this.studentId = studentId;
    }

    public Integer getYearEnrolled() {
        return yearEnrolled;
    }

    public void setYearEnrolled(Integer yearEnrolled) {
        this.yearEnrolled = yearEnrolled;
    }

    public Schedule getWeeklySchedule() {
        return weeklySchedule;
    }

    public void setWeeklySchedule(Schedule weeklySchedule) {
        this.weeklySchedule = weeklySchedule;
    }

    public Transcript getTranscript() {
        return transcript;
    }

    public void setTranscript(Transcript transcript) {
        this.transcript = transcript;
    }

    public Advisor getAdvisor() {
        return advisor;
    }

    public void setAdvisor(Advisor advisor) {
        this.advisor = advisor;
    }

    public Integer getSemester() {
        return semester;
    }

    public void setSemester(Integer semester) {
        this.semester = semester;
    }
}

File: src/python/src/main/marmara/model/Student.py
from src.main.marmara.model.Person import Person


class Student(Person):
    def __init__(self, student_id, year_enrolled, weekly_schedule, transcript, advisor, semester):
        self.student_id = student_id
        self.year_enrolled = year_enrolled
        self.weekly_schedule = weekly_schedule
        self.transcript = transcript
        self.advisor = advisor
        self.semester = semester

    def add_currently_taken_course_to_transcript(self, course):
        self.transcript.add_currently_taken_course(course)

    def add_not_taken_course_to_transcript(self, course):
        self.transcript.add_not_taken_course(course)

    def add_failed_courses_to_transcript(self, course):
        self.transcript.add_failed_course(course)

    def remove_from_transcript_not_taken_course(self, course):
        self.transcript.remove_not_taken_course(course)

    def remove_from_transcript_failed_course(self, course):
        self.transcript.remove_failed_course(course)

    def add_course_to_schedule(self, course):
        self.weekly_schedule.add_course(course)

    def remove_from_schedule(self, course):
        self.weekly_schedule.remove_course(course)

    def calculate_gpa(self):
        return self.transcript.calculate_gpa()

    def __str__(self):
        return (
            f"Student(student_id={self.student_id}, "
            f"year_enrolled={self.year_enrolled}, "
            f"weekly_schedule={self.weekly_schedule}, "
            f"transcript={self.transcript}, "
            f"advisor={self.advisor}, "
            f"semester={self.semester})"
        )

    @property
    def student_id(self):
        return self.student_id

    @student_id.setter
    def student_id(self, student_id):
        self.student_id = student_id

    @property
    def year_enrolled(self):
        return self.year_enrolled

    @year_enrolled.setter
    def year_enrolled(self, year_enrolled):
        self.year_enrolled = year_enrolled

    @property
    def weekly_schedule(self):
        return self.weekly_schedule

    @weekly_schedule.setter
    def weekly_schedule(self, weekly_schedule):
        self.weekly_schedule = weekly_schedule

    @property
    def transcript(self):
        return self.transcript

    @transcript.setter
    def transcript(self, transcript):
        self.transcript = transcript

    @property
    def advisor(self):
        return self.advisor


